lib/handlebars/compiler/compiler.js

```diff
@@ -323,18 +323,28 @@ Compiler.prototype = {
 
   Hash: function(hash) {
     let pairs = hash.pairs,
-        i = 0,
+        i = 0, splat = null,
         l = pairs.length;
 
     this.opcode('pushHash');
-
     for (; i < l; i++) {
+
+      if (pairs[i].type === 'Splat') {
+        if (splat !== null) {
+          throw new Exception('Multiple splats are not supported yet');
```

Comment:
We would also want to do some special casing there for these cases:

1. No splat (Do not call `this.splat`)
2. Only splat (Directly assign the splat path to the hash field)
3. Splat in the first field (will want to pass an empty object to avoid mutating the splat field, i.e. `this.splat({}, rest)`